### ISSUE TITLE
Longhorn is inconsistent with volumes smaller than 1Gi

### ISSUE THREAD

**Issue by dbaker-rh** [longhorn/longhorn#1850]
**Describe the bug**

Longhorn web front end honors disk volumes smaller than 1Gi.  "kubectl create" does not.

Reference to https://github.com/longhorn/longhorn/issues/357 that opened this as a question where the 1Gi minimum size was discussed.

As it stands, longhorn *WILL* create a volume smaller than 1Gi, but from my testing only via web ui.

**To Reproduce - kubectl fails**

````
$ kubectl create -f - <<EOF
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: small-vol
  namespace: default
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: longhorn
  resources:
    requests:
      storage: "0.01Gi"

EOF
````

and then,

````
$ kubectl -n default get pvc
NAME        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
small-vol   Bound    pvc-0b30f658-affa-4342-b0d6-724c49779d82   1Gi        RWO            longhorn       15s
````

Note the Capacity is shown as 1Gi.  Any value smaller than 1Gi rounds up to 1Gi.  Any value over 1Gi is taken as-is.

By comparison, the same "kubectl create ...", but with "storageClassName: local-path" will correctly create a small volume.  With local-path, "storage: 50Mi" works in addition to the approximation "Storage: 0.05Gi".

**To Reproduce - frontend succeeds**

Repro via front end:
- Volume
- Create Volume
- Name: test
- Size: 0.01Gi
- No: 3
- Frontend: Block Device
--> OK

... 

Detached | test | 12 Mi | a few seconds ago
-- | -- | -- | --

Proceeding to manually create PV/PVC and attach to a host via the frontend causes the size to remain 12Mi.

**Expected behavior**

Both methods should be able to create a volume smaller than 1Gi.

**Workaround**

Manually create volumes in the webui, and never via kubectl

**Log**

n/a

**Environment:**
 - Longhorn version: longhornio/longhorn-engine:v1.0.2
 - Kubernetes version: v1.18.9+k3s1
 - Node OS type and version: k3OS v0.11.1

**Additional context**

I've poked about the code in longhorn, longhorn-engine, longhorn-manager and not been able to find reference to anywhere that the volume size might be being rounded up.

A 1Mi volume is (imho) unreasonably small.  The frontend already states that it'll take no more than 2 decimal places.  It seems reasonable that 10Mi might be a workable minimum volume size regardless of how it's created.

**Comment by yasker** [comment]
It was rounded up in https://github.com/longhorn/longhorn-manager/blob/bbfce50e68312cbf0fd5459213738e958851385e/csi/controller_server.go#L138 .

I will close this issue as a duplicate of #357. We will try to address #357 in the upcoming v1.1.0 release.

**Comment by yasker** [comment]
Btw, 10MiB limit sounds good.

**Comment by dbaker-rh** [comment]
Not sure if I should re-open this, or open a new issue.

I've been running with the master branch for a while and bumped into a problem with the new changes.

To reproduce:
- kubectl create .... to create a 256Mi volume.  Use it, then unmount.
- In the longhorn ui, try to "Expand Volume".  The Size gets prefilled as 256Gi.
- If I attempt to resize to "0.5 Gi", the error shows, "Size should be larger than 256Gi".

TLDR - I created a 256Meg volume.  It cannot be resized to any smaller than 256Gig.

**Comment by yasker** [comment]
@dbaker-rh Thanks for the reporting. Can you open a new issue? Likely a UI bug.

**Comment by dbaker-rh** [comment]
Will do.  Thanks.
